src/polymer/analyze-properties.ts

```diff
@@ -150,7 +147,21 @@ export function analyzeProperties(
       prop.readOnly = true;
     }
 
-    prop.type = esutil.CLOSURE_CONSTRUCTOR_MAP.get(prop.type!) || prop.type;
+    // If we've already got a type, then it was found in the jsdocs, and is
+    // canonical. Otherwise we can infer it now from the attribute
+    // deserializer.
+    if (!prop.type && prop.attributeType) {
```

Comment:
do explicit comparisons and don't rely on ToBoolean conversion

Reply:
Done.